lib/scrabble.rb

```diff
@@ -15,4 +13,81 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+
+  def score(word)
+    if word == nil
+      return 0
+    elsif word.is_a? Integer
+      return 0
+    else
+      data = word.upcase.split("")
+      score_conversion_string(data)
+    end
+  end
+
+  def score_conversion_string(data)
+    if data == []
+      return 0
+    else
+      data.map do |letter|
```

Comment:
I like that you're using `#map` & not `#each`! This block is also used in the method below. Try to refactor this out to make increase dry & clean up method length.